front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -105,48 +119,58 @@ export default function AddOrEditScenarioModal({
   const createScenario = async () => {
     if (!currentScenario.name || !currentScenario.infra_id) {
       setDisplayErrors(true);
-    } else {
-      try {
-        const result = await post(`${rootURI}`, { ...currentScenario, infra_id: infraID });
-        dispatch(updateScenarioID(result.id));
-        navigate('/operational-studies/scenario');
-        closeModal();
-      } catch (error) {
-        console.error(error);
-      }
+    } else if (infraID && projectID && studyID && currentScenario && infraID) {
+      postScenario({
+        projectId: projectID,
+        studyId: studyID,
+        scenarioRequest: { ...(currentScenario as ScenarioRequest), infra_id: infraID },
+      })
+        .unwrap()
+        .then(({ id }) => {
+          dispatch(updateScenarioID(id));
+          navigate('/operational-studies/scenario');
+          closeModal();
+        })
+
+        .catch((error) => console.error(error));
     }
   };
 
   const updateScenario = async () => {
     if (!currentScenario.name) {
       setDisplayErrors(true);
-    } else if (scenario) {
-      try {
-        await patch(`${rootURI}${scenario.id}/`, currentScenario);
-        if (getScenarioTimetable) getScenarioTimetable(true);
-        closeModal();
-      } catch (error) {
-        console.error(error);
-      }
+    } else if (scenario && projectID && studyID && scenario.id) {
+      patchScenario({
+        projectId: projectID,
+        studyId: studyID,
+        scenarioId: scenario.id,
+        scenarioPatchRequest: currentScenario,
+      })
+        .unwrap()
+        .then(() => {
+          if (getScenarioTimetable) getScenarioTimetable(true);
+          closeModal();
+        })
+        .catch((error) => console.error(error));
     }
   };
 
   const deleteScenario = async () => {
-    if (scenario) {
-      try {
-        await deleteRequest(`${rootURI}${scenario.id}/`);
-        dispatch(updateScenarioID(undefined));
-        navigate('/operational-studies/study');
-        closeModal();
-        dispatch(
-          setSuccess({
-            title: t('scenarioDeleted'),
-            text: t('scenarioDeletedDetails', { name: scenario.name }),
-          })
-        );
-      } catch (error) {
-        console.error(error);
-      }
+    if (scenario && projectID && studyID && scenario.id) {
+      deleteScenarioRTK({ projectId: projectID, studyId: studyID, scenarioId: scenario.id })
+        .unwrap()
+        .then(() => {
+          dispatch(updateScenarioID(undefined));
+          navigate('/operational-studies/study');
+          closeModal();
+          dispatch(
+            setSuccess({
+              title: t('scenarioDeleted'),
+              text: t('scenarioDeletedDetails', { name: scenario.name }),
+            })
+          );
+        })
+        .catch((error) => console.error(error));
```

Comment:
Same: if the deletion fails, we should inform the user

Reply:
Done !!